src/lib/media-import/status.ts

```diff
@@ -271,44 +273,120 @@ ${ maybeExitPrompt }
 			void checkStatus( IMPORT_MEDIA_PROGRESS_POLL_INTERVAL );
 		} );
 
+	async function exportFailureDetails(
+		fileErrors: Maybe< AppEnvironmentMediaImportStatusFailureDetailsFileErrors >[]
+	) {
+		const formattedData = buildFileErrors( fileErrors, exportFileErrorsToJson );
+		const errorsFile = `media-import-${ app.name ?? '' }-${ Date.now() }${
+			exportFileErrorsToJson ? '.json' : '.txt'
+		}`;
+		try {
+			await writeFile( errorsFile, formattedData );
+			progressTracker.suffix += `${ chalk.yellow(
+				`⚠️  All errors have been exported to ${ chalk.bold( resolve( errorsFile ) ) }`
+			) }`;
+		} catch ( writeFileErr ) {
+			progressTracker.suffix += `${ chalk.red(
+				`Could not export errors to file\n${ ( writeFileErr as Error ).message }`
+			) }`;
+		}
+	}
+
+	async function fetchFailureDetails( failureDetailsUrl: string ) {
+		progressTracker.suffix += `
+=============================================================
+Downloading errors details from ${ failureDetailsUrl }...
+\n`;
+		try {
+			const response = await fetch( failureDetailsUrl );
+			return ( await response.json() ) as AppEnvironmentMediaImportStatusFailureDetailsFileErrors[];
+		} catch ( err ) {
+			progressTracker.suffix += `${ chalk.red(
+				`Could not download file import errors report\n${ ( err as Error ).message }`
+			) }`;
+			throw err;
+		}
+	}
+
+	async function promptFailureDetailsDownload( failureDetailsUrl: string ) {
+		progressTracker.suffix += `${ chalk.yellow(
+			`⚠️  Error details can be found on ${ chalk.bold(
+				failureDetailsUrl
+			) } ${ chalk.italic.yellow(
+				'(This link will be valid for the next 15 minutes. The report is retained for 7 days since the import was run.)'
+			) }. `
+		) }\n`;
+		progressTracker.print( { clearAfter: true } );
+
+		const failureDetails = await prompt( {
+			type: 'confirm',
+			name: 'download',
+			message: 'Download file import errors report now?',
+		} );
+
+		if ( ! failureDetails.download ) {
+			return;
+		}
+
+		const failureDetailsErrors = await fetchFailureDetails( failureDetailsUrl );
+		await exportFailureDetails( failureDetailsErrors );
+	}
+
+	function linkExpiredOutput( results: AppEnvironmentMediaImportStatus ) {
+		if (
+			results.filesTotal &&
+			results.filesProcessed &&
+			results.filesTotal !== results.filesProcessed
+		) {
+			const errorsFound = results.filesTotal - results.filesProcessed;
+			progressTracker.suffix += `${ chalk.yellow(
+				`⚠️  ${ errorsFound } error(s) were found. File import errors report link expired.`
+			) }`;
+		}
+	}
+
+	async function exportLocallyOutput(
```

Comment:
What's the reason for wrapping `exportFailureDetails` in another function?

Reply:
```suggestion
	async function printFailureDetails(
```

also could we rename this?

Reply:
> What's the reason for wrapping exportFailureDetails in another function?

Because it's called by two different points in the application. I didn't want to duplicate code and also keep a function that does more than one job.